docs/design/fabric-v2+/fpc-lifecycle-v2-new.puml

```diff
@@ -0,0 +1,139 @@
+    - the approveformyorg and commit ops are org-specific
+    So the approve stores data for an FPC chaincode installed locally,
+    and the commit checks for data of a locally-approved FPC chaincode.    
+end note
+FPC_Peer1_Wrapper -> FPC_Peer1_Wrapper : retrieve cc_language from package
+note over FPC_Peer1_Wrapper
+    Queryinstalled is redudant, yet consistent with implementation.    
+end note
+FPC_Peer1_Wrapper -> Peer1 ++: lifecycle chaincode queryinstalled
+return packageId
+FPC_Peer1_Wrapper -> FPC_Peer1_Wrapper : if cc_language==fpc then store packageId
+return packageId /' Admin -> FPC_Peer1_Wrapper '/
+end /' group install '/
+
+ref over Admin, FPC_Peer2_Wrapper, Peer2
+    install chaincode on Peer2
+end /' ref '/
+
+group /' approve '/
+Admin -> FPC_Peer1_Wrapper ++: lifecycle chaincode approveformyorg
+FPC_Peer1_Wrapper -> Peer1 ++: lifecycle chaincode approveformyorg
+note right
+    **Chaincode definition**
+    Name: cc_name, Version: mrenclave,
+    Sequence: sequence_number, Endorsement Policy: (2of3),
+    Plugins: <custom endorsement/validation plugins>
+end note
+Peer1 -> Orderer ++: approve transaction
+return transaction committed /' Peer1 -> Orderer '/
+return /' FPC_Peer1_Wrapper -> Peer1 '/
+note over FPC_Peer1_Wrapper
+    Bookkeping operations.
+    Note: packageId is an input parameter of approveformyorg
+end note
+FPC_Peer1_Wrapper -> FPC_Peer1_Wrapper : if packageId was stored then store cc_namie, cc_version
+return /' Admin -> FPC_Peer1_Wrapper '/
+
+Channel -> Channel : approvals from other parties
+note over Channel
+    Other participants must approve the chaincode definition,
+    and commit the corresponding transaction,
+    according to the LifecycleEndorsement policy.
+    So the party here has to loop to check for the approvals.
+end note
+
+end /' group approve '/
+
+loop until enough approvals
+	Admin -> Peer1 : lifecycle chaincode checkcommitreadiness
+note right
+    **Chaincode definition**
+    (implemented to match the approved one)
+end note
+    return
+end
+
+group commit
+Admin -> FPC_Peer1_Wrapper ++:lifecycle chaincode commit
+FPC_Peer1_Wrapper -> Peer1 ++: lifecycle chaincode commit
+note right
+    **Chaincode definition**
+    (implemented to match the one that is ready to be committed)
+end note
+Peer1 -> Orderer ++: commit transaction
+return transaction committed /' Peer1 -> Orderer '/
+return /' FPC_Peer1_Wrapper -> Peer1 '/
+note over FPC_Peer1_Wrapper
+    Commit has succeeded.
+    Now, check if this fpc cc was previously approved.
+    If so, perform some invocations.
+end note
+FPC_Peer1_Wrapper -> FPC_Peer1_Wrapper : if cc_name, cc_version not stored then return
+FPC_Peer1_Wrapper -> Peer1 ++: invoke chaincode __setup
+Peer1 -> Orderer ++: invoke transaction
+return transaction committed
+return /' FPC_Peer1_Wrapper -> Peer1 '/
+FPC_Peer1_Wrapper -> Peer1 ++: query chaincode __getEnclavePk
+return /' FPC_Peer1_Wrapper -> Peer1 '/
```

Comment:
This does not align well with the new enclave registration protocol I just reviewed.

Reply:
07b4ce2a989f777ea4f0b16722ca2a6274ce6a15